torch/nn/_functions/rnn.py

```diff
@@ -271,7 +271,8 @@ def forward(input, weight, hx, batch_sizes):
             cx = None
 
         handle = cudnn.get_handle()
-        dropout_ts = cudnn.rnn.init_dropout_state(torch.uint8, torch.device('cuda'), dropout,
+        with torch.cuda.device(input.get_device()):
+            dropout_ts = cudnn.rnn.init_dropout_state(torch.uint8, torch.device('cuda'), dropout,
```

Comment:
It's kind of inconvenient, because the actual allocation happens in `cudnn/Descriptors.h`:

```
    state = ty.tensor({static_cast<int64_t>(state_size)});
```

And there isn't any way to say, from the C++ API, what device you want when you allocate a tensor this way:

```
  virtual Tensor tensor(Storage & storage, int64_t storageOffset, IntList size, IntList stride={}) const;
  virtual Tensor tensor(IntList size) const;
  virtual Tensor tensor(IntList size, IntList stride) const;
  virtual Tensor tensor() const;
```

This should all be fixed, but I wouldn't let it block a perfectly good bugfix.